metis-technical-metadata-generation/src/main/java/eu/europeana/metis/technical/metadata/generation/utilities/MediaExtractorForFile.java

```diff
@@ -155,6 +194,18 @@ private void submitResource(final FileStatus fileStatus, final Set<Integer> nonR
     });
   }
 
+  private void thumbnailUpload(String resourceUrl) {
+    final TechnicalMetadataWrapper technicalMetadataWrapper = mongoDao
+        .getTechnicalMetadataWrapper(resourceUrl);
+    if (CollectionUtils.isEmpty(technicalMetadataWrapper.getThumbnailWrappers())) {
```

Comment:
Avoid dependency on spring framework here? Just check null and .isEmpty().

Reply:
Sure although it's a transient dependency